File: src/model/Clube.java
package model;

import java.time.LocalDate;
import java.util.ArrayList;

public class Clube {
    private String nome;
    private LocalDate criacao;
    private ArrayList<Atleta> atletas;
    private ArrayList<Partida> partidas;

    public Clube() {
        this.atletas = new ArrayList<>();
        this.partidas = new ArrayList<>();
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public LocalDate getCriacao() {
        return criacao;
    }

    public void setCriacao(LocalDate criacao) {
        this.criacao = criacao;
    }

    public ArrayList<Atleta> getAtletas() {
        return atletas;
    }

    public ArrayList<Partida> getPartidas() {
        return partidas;
    }

    public void adicionarAtleta(Atleta atleta) {
        this.atletas.add(atleta);
    }
    public void adicionarPartida(Partida partida) {
        this.partidas.add(partida);
    }

    public void exibirAtletas() {
        System.out.println("Atletas: ");
        for (Atleta atleta : atletas) {
            System.out.println(atletas);
        }
    }

    public void exibirPartidas() {
        System.out.println("Partidas: ");
        for (Partida partida : partidas) {
            System.out.println(partida);
        }
    }

}

File: sgfc_too/lib/Clube.dart
import '../atleta.dart';
import '../partida.dart';
import '../funcionario.dart';

class Clube {
  String _nome = '';
  DateTime? _criacao;
  final List<Atleta> _atletas = [];
  final List<Partida> _partidas = [];
  final List<Funcionario> _funcionarios = [];

  // Getters e setters
  String get nome => _nome;
  set nome(String value) => _nome = value;

  DateTime? get criacao => _criacao;
  set criacao(DateTime? value) => _criacao = value;

  List<Atleta> get atletas => _atletas;
  List<Partida> get partidas => _partidas;
  List<Funcionario> get funcionarios => _funcionarios;

  // Métodos para adicionar
  void adicionarAtleta(Atleta atleta) {
    _atletas.add(atleta);
  }

  void adicionarPartida(Partida partida) {
    _partidas.add(partida);
  }

  void adicionarFuncionario(Funcionario funcionario) {
    _funcionarios.add(funcionario);
  }

  // Métodos para exibir
  void exibirAtletas() {
    print('Atletas:');
    for (var atleta in _atletas) {
      print(atleta);
    }
  }

  void exibirPartidas() {
    print('Partidas:');
    for (var partida in _partidas) {
      print(partida);
    }
  }

  void exibirFuncionarios() {
    print('Funcionários:');
    for (var funcionario in _funcionarios) {
      print(funcionario);
    }
  }
}
